datalad/interface/annotate_paths.py

```diff
@@ -124,21 +125,28 @@ def annotated2content_by_ds(annotated, refds_path, path_only=False):
                 # content rather then the dataset itself
                 # in both cases we want to process this part as part
                 # of the same dataset, and not any potential parent
-                toappendto = content_by_ds.get(r['path'], [])
-                toappendto.append(r['path'] if path_only else r)
-                content_by_ds[r['path']] = toappendto
+                toappendto = content_by_ds.get(r_path, [])
+                toappendto.append(r_path if path_only else r)
+                content_by_ds[r_path] = toappendto
             if parentds and refds_path and \
                     path_startswith(parentds, refds_path):
                 # put also in parentds record if there is any, and the parent
                 # is underneath or identical to the reference dataset
                 toappendto = content_by_ds.get(parentds, [])
-                toappendto.append(r['path'] if path_only else r)
+                toappendto.append(r_path if path_only else r)
                 content_by_ds[parentds] = toappendto
         else:
             # files and dirs
             # common case, something with a parentds
             toappendto = content_by_ds.get(parentds, [])
-            toappendto.append(r['path'] if path_only else r)
+            if path_only:
+                r_path_src = r.get('path_src')
```

Comment:
so it seems that nowhere in the code path_only=True is used
```shell
$> git grep 'path_only=[^F]'
```
I would be happy to remove this option/logic if @mih agrees to such lobotomization ;-)

Reply:
This option was needed for compat with legacy code. If it is no longer needed, I would be happy to see it go.

Reply:
cool, then I will do it in this PR so we do not forget. thanks